psi4/src/export_wavefunction.cc

```diff
@@ -129,6 +129,10 @@ void export_wavefunction(py::module& m) {
         .def("get_basisset", &Wavefunction::get_basisset, "Returns the requested auxiliary basis.")
         .def("set_basisset", &Wavefunction::set_basisset, "Sets the requested auxiliary basis.")
         .def("energy", &Wavefunction::reference_energy, "Returns the Wavefunctions energy.")
+        .def("dipole_gradient", &Wavefunction::dipole_gradient,
+             "Returns the Wavefunctions dipole derivative w.r.t. nuclear perturbations (a.u.) as a 3N x 3 matrix.")
+        .def("set_dipole_gradient", &Wavefunction::set_dipole_gradient,
```

Comment:
Can we hold these in the `get/set_array` generics?

Reply:
if it helps the decision making, I think the named arrays (energy, gradient, hessian, dipole_gradient?) are (1) the things we set `CURRENT` for and (2) the things that, on the wfn, should always be a synched level of theory. A related question is whether after qcvar reform completed, E/G/H on wfn goes away and qcvars on wfn w/ a complement of `CURRENT`s take over or whether E/G/H on wfn continues and `CURRENT`s dropped altogether.